src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -4470,9 +4469,9 @@ public void visitSelect(JCFieldAccess tree) {
             // If the qualified item is not a type and the selected item is static, report
             // a warning. Make allowance for the class of an array type e.g. Object[].class)
             if (!sym.owner.isAnonymous()) {
-                chk.warnStatic(tree, Warnings.StaticNotQualifiedByType(sym.kind.kindName(), sym.owner));
+                chk.lint.logIfEnabled(tree, LintWarnings.StaticNotQualifiedByType(sym.kind.kindName(), sym.owner));
```

Comment:
This is an odd one - the rest of `Attr` uses `env.info.lint` but if I do that here I get some NPE in some unrelated JVM test. I wonder how robust using two different lints is... but for now I'll keep compatibility with what we have.